src/store.js

```diff
@@ -0,0 +1,29 @@
+import {createStore, applyMiddleware, compose} from 'redux';
+import rootReducer from './reducers';
+import ReduxThunk from 'redux-thunk/';
+import {persistStore, persistReducer} from 'redux-persist';
+import storage from 'redux-persist/lib/storage';
+import autoMergeLevel1 from 'redux-persist/lib/stateReconciler/autoMergeLevel1';
+
+const persistConfig = {
+    key: 'root',
+    storage: storage,
+    // autoMergeLevel1 is the default anyway, but I want to be explicit
+    stateReconciler: autoMergeLevel1,
+    whitelist: ['auth', 'map']
+};
+
+const persistedReducer = persistReducer(persistConfig, rootReducer);
+
+export const store = createStore(
+    persistedReducer,
+    {},
+    compose(
+        applyMiddleware(ReduxThunk)
+    ));
+
+export const persistor = persistStore(store);
```

Comment:
Last thing we do is also export the `persistor` so that it can be used across the app. (Remember this is the object we grabbed in both `App.js` to pass to `PersistGate` as a prop, and we also used in the `AuthActions` to call `persistor.purge()` and clear out all `AsyncStorage` data associated with our `redux-persist` work.